docs/edit-yaml.md

```diff
@@ -0,0 +1,121 @@
+```yaml
+steps:
+    - files:
+        .github/workflows/deploy.yml:
+            action: edit-yaml
+            changes:
+                - remove:
+                    properties: {
+                        on: ['pull_request']
+                    }
+                - remove:
+                    properties: {
+                        jobs: {
+                            Publish: ['steps']
+                        }
+                    }
+```
+
+When using `path`, the location of the property to alter is provided in an array format. If no `value` is provided, the property is removed. A `value` or `values` can be provided to specify removing specific values from an existing array. Using `path` also allows the specification of an array index within the path - making it easy to target properties of arrays.
+
+The example below accomplishes the same as the example above, but using the `path` method.
+
+```yaml
+steps:
+    - files:
+        .github/workflows/deploy.yml:
+            action: edit-yaml
+            changes:
+                - remove:
+                    path: ['on']
+                    value: 'pull_request'
+                - remove:
+                    path: ['jobs', 'Publish', 'steps']
+```
+
+### `update`
+Update is used to add a new key/value, alter an existing property, or add to an array.
+
+When using `properties`, the keys provided are updated, but any other keys and values within the objecte are preserved. This method allows for modifiying multipe properties utilizing a single update entry. This is useful when updating dependencies, as these can easily be copy/pasted into a new patch file. When altering an existing array using `properties`, the existing array will *always* be appended to.
+
+The example below updates the "Publish" job by adding a new property `timeout-minutes` (keeping the rest of the properties in place), and adds a new step to the `steps` array.
+```yaml
+steps:
+    - files:
+        .github/workflows/deploy.yml:
+            action: edit-yaml
+            changes:
+                - update:
+                    properties: {
+                        jobs: {
+                            Publish: {
+                                timeout-minutes: 10,
+                                steps: [{
+                                    name: New step,
+                                    uses: some/action@v1
+                                }]
+                            }
+                        }
+                    }
+```
+
+Using the `path` method is more verbose, as each path and value must be specified in a separate `update` statement. The verboseness of this method provides more control however, allowing one to specify how to add to existing arrays by providing a `modifier`. The `modifier` currently supports `set`, `append` and `prepend` - `set` is the default if none is provided and will set the property to the provided value (or values), replacing any existing value(s).
```

Comment:
> The verboseness of this method provides more control however, allowing one to specify how to add to existing arrays by providing a `modifier`.

Should 'allowing' be 'allows' here?
OR
provides more control ~~however~~, allowing one

Reply:
Yea, this does a read a bit awkwardly - I like your second suggestion, I'll edit to something along those lines.